crates/bevy_ecs/src/error/handler.rs

```diff
@@ -10,12 +10,19 @@ pub struct SystemErrorContext {
     pub last_run: Tick,
 }
 
-/// The default systems error handler stored as a resource in the [`World`](crate::world::World).
-pub struct DefaultSystemErrorHandler(pub fn(BevyError, SystemErrorContext));
+/// The error handler of last resort used for [`bevy_ecs::error::Result`]s returned by systems, commands and observers,
+/// when an error is not otherwise handled.
+///
+/// This is stored as a resource in the [`World`](crate::world::World),
+/// and defaults to panicking if not set.
+///
+/// See [`bevy_ecs::error`] for more information on error handling,
+/// and [`bevy_ecs::error::handler`] for an assortment of built-in error handlers.
+pub struct FallbackErrorHandler(pub fn(BevyError, SystemErrorContext));
```

Comment:
I see the appeal of removing `System`, but as a user if I was going to change this, I would ask myself "how do I set the default bevy error handler". I also think it "fits". People tend to consider defaults "overridable".